### PR TITLE
feat(azure): subscription name aliases

### LINKED ISSUE #4448: Add support for Azure subscription aliases
#### Is your feature request related to a problem? Please describe.

No, it is not related to a problem.

#### Describe the solution you'd like

Several other Starship modules offer the ability to supply aliases for some portion(s) of the content being displayed by that module. For example, the Google Cloud module supports aliasing for regions and projects, via the `regions_aliases` and `project_aliases` configuration parameters. I'd like to see similar support for Azure subscriptions, so that we can provide shorter aliases for the (sometimes) long Azure subscription names.

#### Describe alternatives you've considered

I don't know that there _are_ any other alternatives. It may be possible to manually edit the `azureProfile.json` file by which Starship retrieves its information, but I haven't tested that and suspect that any such changes would get overwritten upon the next re-authentication interval.

### PR DESCRIPTION
**Pull request by marcybelardo**
#### Description

- Added `subscription_aliases` field for the `azure` module.

- Similar to the `gcloud` module, takes `[azure.subscription_aliases]` from `starship.toml` and replaces the subscription name. If there is no alias, defaults to the normal name.

#### Motivation and Context

Closes #4448

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Review by davidkna — COMMENTED** [review]
This looks good to me, but please update the module documentation.

File: docs/config/README.md [hunk 1]
```diff
@@ -745,7 +755,7 @@ error_symbol = '[➜](bold red) '
 # ~/.config/starship.toml
 
 [character]
-vicmd_symbol = '[V](bold green) '
+vimcmd_symbol = '[V](bold green) '
```
Comment (davidkna) [inline docs/config/README.md:758]: I think you added some extra changes.

Comment (marcybelardo) [reply docs/config/README.md]: Sorry, some things stuck over from a merge, reverted those back just now

**Review by davidkna — APPROVED** [review]

**Comment by davidkna** [comment]
Thanks for the contribution @marcybelardo

**Pull request merged**